vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -607,284 +616,202 @@ facebook::jsi::Runtime::ScopeState *ChakraRuntime::pushScope() {
 
 void ChakraRuntime::popScope(Runtime::ScopeState *state) {
   assert(state == nullptr);
-  checkException(JsCollectGarbage(m_runtime), "JsCollectGarbage");
+  ThrowUponJsError(JsCollectGarbage(m_runtime), "JsCollectGarbage");
 }
 
 bool ChakraRuntime::strictEquals(
     const facebook::jsi::String &a,
     const facebook::jsi::String &b) const {
-  bool result;
-  JsStrictEquals(stringRef(a), stringRef(b), &result);
-  return result;
+  return CompareJsValues(GetChakraObjectRef(a), GetChakraObjectRef(b));
 }
 
 bool ChakraRuntime::strictEquals(
     const facebook::jsi::Object &a,
     const facebook::jsi::Object &b) const {
-  bool result;
-  JsStrictEquals(objectRef(a), objectRef(b), &result);
-  return result;
+  return CompareJsValues(GetChakraObjectRef(a), GetChakraObjectRef(b));
 }
 
 bool ChakraRuntime::strictEquals(
-    const facebook::jsi::Symbol &,
-    const facebook::jsi::Symbol &) const {
-  throw facebook::jsi::JSINativeException("Not implemented!");
+    const facebook::jsi::Symbol &a,
+    const facebook::jsi::Symbol &b) const {
+  return CompareJsValues(GetChakraObjectRef(a), GetChakraObjectRef(b));
 }
 
 bool ChakraRuntime::instanceOf(
-    const facebook::jsi::Object &o,
-    const facebook::jsi::Function &f) {
-  bool res;
-  checkException(JsInstanceOf(objectRef(o), objectRef(f), &res));
-  return res;
+    const facebook::jsi::Object &obj,
+    const facebook::jsi::Function &func) {
+  bool result;
+  ThrowUponJsError(
+      JsInstanceOf(GetChakraObjectRef(obj), GetChakraObjectRef(func), &result),
+      "JsInstanceOf");
+  return result;
 }
 
 #pragma endregion Functions_inherited_from_Runtime
 
-facebook::jsi::Value ChakraRuntime::createValue(JsValueRef value) const {
-  JsValueType type;
-  JsGetValueType(value, &type);
+facebook::jsi::Value ChakraRuntime::ToJsiValue(ChakraObjectRef &&ref) {
+  JsValueType type = GetValueType(ref);
 
   switch (type) {
-    case JsUndefined:
+    case JsUndefined: {
       return facebook::jsi::Value();
-
-    case JsNull:
+      break;
+    }
+    case JsNull: {
       return facebook::jsi::Value(nullptr);
-
+      break;
+    }
     case JsNumber: {
-      double doubleValue;
-      JsNumberToDouble(value, &doubleValue);
-      facebook::jsi::Value val(doubleValue);
-      return val;
+      double number;
+      ThrowUponJsError(JsNumberToDouble(ref, &number), "JsNumberToDouble");
+      return facebook::jsi::Value(number);
+      break;
     }
-
     case JsString: {
-      std::string utf8str = JSStringToSTLString(value);
-      return facebook::jsi::String::createFromUtf8(
-          *const_cast<facebook::jsi::Runtime *>(
-              reinterpret_cast<const facebook::jsi::Runtime *>(this)),
-          reinterpret_cast<const uint8_t *>(utf8str.c_str()),
-          utf8str.size());
+      return facebook::jsi::Value(
+          *this, MakePointer<facebook::jsi::String>(std::move(ref)));
```

Comment:
>MakePointer [](start = 17, length = 11)

(Nit) This changes the semantic (before we used to make a copy of string); are there runtime implications / assumptions existing code made?

Reply:
The change to move semantics is intentional and should not have any runtime implications as far as I know. (With one exception, this function is always used to wrap up a newly obtained JsValueRef in a jsi::Value. The one exception is in HostFunctionCall, butI haven't seen any runtime issues when its code is exercised during E2ETest build.)

---
In reply to: [336277187](https://github.com/microsoft/react-native-windows/pull/3445#discussion_r336277187) [](ancestors = 336277187)